gateway/scim_ethermab.py

```diff
@@ -52,8 +56,20 @@ def ethermab_create_device(request,device_id):
         raise DeviceExists
 
     api=init_ise()
+    endpoint=None
     if api:
-        api.endpoint.create_endpoint(mac=device_mac_address)
+        try:
+            endpoint = api.endpoint.get_endpoint_by_name(device_mac_address)
+        except ApiError as e:
+            if e.status_code != 404:
+                raise ISEError(e.description) from e
+            # if it is a 404, device is in ISE- let it be
```

Comment:
I'm not sure I understand this comment. If the device exists, why do we create the endpoint in the next line? Or does it mean that the "device is _not_ in ISE"?

Reply:
The issue here is that the mac address may already be listed, and so on create a 404 will be generated.

Reply:
Yeah, this code has to stay.  endpoint will be None if the error is raised, and we only want to ignore it if it's because the object doesn't exist.  otherwise it's something we didn't expect.

Reply:
Ok got it.